style.md

```diff
@@ -0,0 +1,281 @@
+# Style Guide
+
+The following guidelines are used for code in the lecture notes (i.e., not to the `QuantEcon.jl` package).  See [Julia Style Guide](https://docs.julialang.org/en/v1/manual/style-guide/) for the baseline guidelines, which this supplements.
+
+## Basic Principles
+
+Keep in mind that these lectures are targeted at students with (at most!) some self-taught Matlab or Python experience.  Consequently, we want to ensure that the code is clear and simple to focus on the Economics/Math and only expect them to write code using a simplified subset of Julia features.  Some guiding principles:
+
+1. Assume this may be the **first programming language** students learn
+2. Use **compact, script-style** code, organized into functions only when it is natural.  Best practices for writing packages and expository/exploratory/interactive code can be different.
+3. Keep things as **close to the whiteboard math** as possible, including in the code structure and notation
+4. Ensure that all **code can be copied and pasted without modification** into functions for performance and modification without changes to scoping (e.g. no `local` or `global` ever required)
+5. **Avoid type annotations** unless they are required for dispatching
+6. **Avoid creating custom types** unless it is absolutely necessary
+7. Beyond the [Julia Style Guide](https://docs.julialang.org/en/v1/manual/style-guide/), avoid unnecessary whitespace lines and redundant comments
+8. Don't use fancy features and control flow from Julia - unless it makes the code look closer to the math
+9. Avoid both micro-optimizations and coding patterns that pessimize (i.e. poor performance with no benefit in code clarity)
+
+We want users to be able to say _"the code is clearer than Matlab, and even closer to the math"_.
+
+## Naming Conventions, Comments, etc.
+
+- **Use unicode** where possible in names so that symbols match the math in the document
+- **Be careful** about unicode glyphs and symbols which may not be available in the default REPL, Jupyter, etc. for all platforms.  **TODO** 
```

Comment:
I think we should reword this a bit, because we're saying on the one hand to use this wherever possible (including, as written, for things like `tilde` and `bar` and `hat`), and on the other to be careful.

Reply:
Once we test for the fonts we can.